.github/workflows/compatibility.yml

```diff
@@ -0,0 +1,72 @@
+name: Compatibility - WC, WP and PHP
+
+on:
+  pull_request
+
+jobs:
+  woocommerce-compatibility:
+    name:    WC compatibility
+    runs-on: ubuntu-16.04
+    strategy:
+      fail-fast:    false
+      max-parallel: 10
+      matrix:
+        woocommerce: [ '4.0.0', '4.3.0', '4.4.0', '4.5.0', '4.6.0', '4.7.0', '4.8.0', '4.9.0' ]
+        wordpress:   [ 'latest' ]
+        php:         [ '7.4' ]
+        include:
+          # Edge case: oldest dependencies compatibility
+          - woocommerce: '3.3.0'
+            wordpress:   '4.7'
+            php:         '5.6'
+    env:
+      WP_VERSION: ${{ matrix.wordpress }}
+      WC_VERSION: ${{ matrix.woocommerce }}
+    steps:
+      # clone the repository
+      - uses: actions/checkout@v2
+      # enable dependencies caching
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/composer/
+          key:  ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+      # setup PHP, but without debug extensions for reasonable performance
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools:       composer
+          coverage:    none
+      # run CI checks
+      - run: if [[ "${{ matrix.php }}" == '5.6' ]]; then wget https://phar.phpunit.de/phpunit-5.7.27.phar && mv phpunit-5.7.27.phar phpunit.phar; fi;
+      - run: if [[ "${{ matrix.php }}" == '7.0' ]]; then wget https://phar.phpunit.de/phpunit-6.5.14.phar && mv phpunit-6.5.14.phar phpunit.phar; fi;
```

Comment:
Would the right version in `[[ "${{ matrix.php }}" == '7.0' ]]` be 7.4?

Reply:
By default the test image uses the PHPUnit version specified in the composer file (currently 7.5.20), which requires PHP 7.1 +, the `ifs` are an override to use the latest compatible version of PHPUnit for each PHP version back to 5.6.
When we bump the composer dependency we'll need to add another line here for 7.1 (and for 7.2 if we bump to PHPUnit to the latest version)

Also, at this time the versions matrix only includes PHP 5.6 and 7.4, that's something to change after merging the Linting PR... we should split this into 3 jobs instead of 2, something like this:
- legacy (EOL versions: 7.2- down to the min version supported by Woo)
- stable (7.3, 7.4)
- latest (8.0)

Reply:
Understood. I asked about that `if` because I didn't see any PHP 7.0.

> Also, at this time the versions matrix only includes PHP 5.6 and 7.4, that's something to change after merging the Linting PR... we should split this into 3 jobs instead of 2, something like this:

Sounds good 👍